### ISSUE TITLE
Accessibility: Multiple form field and table header labels are missing in template files

### ISSUE THREAD

**Issue by webmandesign** [woothemes/woocommerce#31210]
**Prerequisites (mark completed items with an [x]):**
- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
Checking various WooCommerce views (such as Cart, Checkout and My Account pages, and widgets) with [WAVE accessibility evaluation tools](https://wave.webaim.org/) produces accessibility issues of missing labels for form fields and table header cells.

**Expected behavior**
All form fields should have labels set with `<label>` tag. This can be hidden with `screen-reader-text` class if needed.
All table `<th>` tags should have descriptive text. This can be hidden wrapped with `<span class="screen-reader-text"></span>` tag if needed.

**Actual behavior**
- Shop table on Cart page has no label text for first 2 table columns.
- Shipping calculator on Cart page is missing form field labels.
- Coupon form on Checkout page is missing form field label.
- Price filter widget is missing form field labels.
- Grouped product add to cart form is missing a form field label.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:
1. Test various template views with [WAVE](https://wave.webaim.org/) (a [browser extension](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh) can be used).

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
WooCommerce 5.9.0, WordPress 5.8.2

**Comment by rodelgc** [comment]
Hello @webmandesign,

Thank you for taking the time to submit this issue and a PR as well! We really appreciate your help.

While this is clearly a great improvement/idea we won’t be able to tackle it in the upcoming weeks. We’re going to add it to our backlog where it will be considered for future releases.
﻿
Stay tuned for updates.

**Comment by webmandesign** [comment]
Hi @rodelgc,

Thank you for the info. It's a bit unfortunate, but I guess I have to wait :)

An alternative approach would be to fix this via a theme, but I'd say that would not be ideal solution especially if the theme does not override WooCommerce templates otherwise.

**Comment by rodelgc** [comment]
@webmandesign the PR you submitted actually could help speed up the fix 👍  
Thank you very much again for actively submitting a solution. Just give the Core team some time to get to your PR.

**Comment by webmandesign** [comment]
For anyone interested, I've released a plugin at https://wordpress.org/plugins/wc-a11y/ fixing these issues now.
